### ISSUE TITLE
[BUG] RewardScaler missing torch / cuda casting

### ISSUE THREAD

**Issue by lettersfromfelix** [takuseno/d3rlpy#212]
**Describe the bug**
When utilizing a reward scaler while training on gpu, I get the error Can't convert cuda:0 device type tensor to xyz. Use Tensor.cpu() to copy the tensor to host memory first

**To Reproduce**
Define reward scaler, perform offline training on GPU, in my case with CQL

**Expected behavior**
Working as with every other scaler

**Additional context**
I've had a look at the other scalers, and noticed that they have a torch.tensor() cast around the variables in question. In my case for the StandardRewardScaler, the following snippet seems to fix the issue:
```
def transform(self, reward: torch.Tensor) -> torch.Tensor:
        assert self._mean is not None and self._std is not None
        mean = torch.tensor(self._mean, dtype=torch.float32, device=reward.device)
        std = torch.tensor(self._std, dtype=torch.float32, device=reward.device)
        return self._multiplier * (reward - mean) / (std + self._eps)

    def reverse_transform(self, reward: torch.Tensor) -> torch.Tensor:
        assert self._mean is not None and self._std is not None
        mean = torch.tensor(self._mean, dtype=torch.float32, device=reward.device)
        std = torch.tensor(self._std, dtype=torch.float32, device=reward.device)
        return reward * (std + self._eps) / self._multiplier + mean
```

**Comment by takuseno** [comment]
@lettersfromfelix Thanks for the issue. This is actually weird because `self._mean` and `self._std` are `float` values because they are scalar numbers. Thus, it won't be complained about which device to use. Also, I confirmed this works in my environment.
```
import d3rlpy
import numpy as np

observations = np.random.random((1000, 1000)).astype(np.float32)
actions = np.random.random((1000, 4)).astype(np.float32)
rewards = np.random.random(1000).astype(np.float32)
terminals = np.zeros(1000)
terminals[-1] = 1.0

dataset = d3rlpy.dataset.MDPDataset(observations, actions, rewards, terminals)
cql = d3rlpy.algos.CQL(batch_size=256, use_gpu=True, reward_scaler="standard")

cql.fit(dataset, n_steps=10000, n_steps_per_epoch=1000)
```
Can you provide a minimal example that I can reproduce your issue on my end?

**Comment by lettersfromfelix** [comment]
Seems like I get the error as soon as I manually initialize the reward_scaler. When I just set the string argument it works fine for me as well.

```
import d3rlpy
import numpy as np

observations = np.random.random((1000, 1000)).astype(np.float32)
actions = np.random.random((1000, 4)).astype(np.float32)
rewards = np.random.random(1000).astype(np.float32)
terminals = np.zeros(1000)
terminals[-1] = 1.0

dataset = d3rlpy.dataset.MDPDataset(observations, actions, rewards, terminals)

mean_rew = np.mean(dataset.rewards, axis=0, keepdims=True)
std_rew = np.std(dataset.rewards, axis=0, keepdims=True)
reward_scaler = d3rlpy.preprocessing.StandardRewardScaler(mean=mean_rew, std=std_rew)

cql = d3rlpy.algos.CQL(use_gpu=True, reward_scaler=reward_scaler)

cql.fit(dataset, n_epochs=1)
```

**Comment by takuseno** [comment]
Yeah, you need to remove `axis=0` and `keepdims=True`.

**Comment by lettersfromfelix** [comment]
You're right, it works that way. Still, it's a topic for the docs, because I've been following this: https://d3rlpy.readthedocs.io/en/v1.1.1/tutorials/preprocess_and_postprocess.html

Plus, even though cuda seems to be fine with that and the training works, my language server (pylance) is hinting me an error in the code now:
```
Argument of type "floating[Any]" cannot be assigned to parameter "mean" of type "float | None" in function "__init__"
  Type "floating[Any]" cannot be assigned to type "float | None"
    "floating[Any]" is incompatible with "float"
    Type cannot be assigned to type "None"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pylance-release/blob/main/DIAGNOSTIC_SEVERITY_RULES.md#diagnostic-severity-rules)
```

**Comment by takuseno** [comment]
Thanks for pointing out the docs error. It seems copy/paste issue. Regarding the type hint, it would be gone if you do cast like this `float(mean)`. Let me close this issue since the issue seems resolved.
